python_modules/libraries/dagster-datahub/dagster_datahub/resources.py

```diff
@@ -8,79 +11,82 @@
     KafkaEmitterConfig,
 )
 from datahub.emitter.rest_emitter import DatahubRestEmitter
+from pydantic import Field as PyField
 
 
-@resource(
-    config_schema={
-        "connection": Field(StringSource, is_required=True, description="Datahub GMS Server"),
-        "token": Field(
-            Noneable(StringSource),
-            default_value=None,
-            is_required=False,
-            description="Personal Access Token",
-        ),
-        "connect_timeout_sec": Field(Noneable(float), default_value=None, is_required=False),
-        "read_timeout_sec": Field(Noneable(float), default_value=None, is_required=False),
-        "retry_status_codes": Field(Noneable(Array(int)), default_value=None, is_required=False),
-        "retry_methods": Field(Noneable(Array(str)), default_value=None, is_required=False),
-        "retry_max_times": Field(Noneable(int), default_value=None, is_required=False),
-        "extra_headers": Field(Noneable(Map(str, str)), default_value=None, is_required=False),
-        "ca_certificate_path": Field(Noneable(str), default_value=None, is_required=False),
-        "server_telemetry_id": Field(Noneable(str), default_value=None, is_required=False),
-        "disable_ssl_verification": Field(bool, default_value=False, is_required=False),
-    }
-)
+class DatahubRESTEmitterResource(ConfigurableResource):
+    connection: str = PyField(description="Datahub GMS Server")
+    token: Optional[str] = PyField(default=None, description="Personal Access Token")
+    connect_timeout_sec: Optional[float] = PyField(default=None)
+    read_timeout_sec: Optional[float] = PyField(default=None)
+    retry_status_codes: Optional[List[int]] = PyField(default=None)
+    retry_methods: Optional[List[str]] = PyField(default=None)
+    retry_max_times: Optional[int] = PyField(default=None)
+    extra_headers: Optional[Dict[str, str]] = PyField(default=None)
+    ca_certificate_path: Optional[str] = PyField(default=None)
+    server_telemetry_id: Optional[str] = PyField(default=None)
+    disable_ssl_verification: bool = PyField(default=False)
```

Comment:
can just do `connect_timeout_sec: Optional[float] = None` for all of these, no?

Reply:
yup 👍